### ISSUE TITLE
PHP 8 - PSR12.Classes.ClassInstantiation.MissingParentheses breaks code

### ISSUE THREAD

**Issue by m-ober** [squizlabs/PHP_CodeSniffer#3140]
**Describe the bug**
Using PHP8 RC1 and PHPCS 3.5.6, the sniff PSR12.Classes.ClassInstantiation.MissingParentheses produces spurious errors and breaks code when using phpcbf.

**Code sample**
```php
function foo()
{
    return new \DatePeriod($dates['starts_at'], new \DateInterval('P1D'), $dates['ends_at']);
}
```

phpcbf will place parentheses after new `new` statements (which is invalid code):

```php
function foo()
{
    return new() \DatePeriod($dates['starts_at'], new() \DateInterval('P1D'), $dates['ends_at']);
}
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
 5 | ERROR | [x] Parentheses must be used when instantiating a new class
   |       |     (PSR12.Classes.ClassInstantiation.MissingParentheses)

```

**Expected behavior**
There is no PSR12 violation in this line. 

**Versions (please complete the following information):**
 - OS: Linux
 - PHP: 8.0 RC1
 - PHPCS: 3.5.6
 - Standard: PSR12

**Additional context**
Problem does not appear with PHP 7.4.

**Comment by m-ober** [comment]
Nvm, seems to be fixed in master.
